lib/ScssCompiler.js

```diff
@@ -0,0 +1,280 @@
+const _ = require('lodash');
+const path = require('path');
+const nodeSassFork = require('@bigcommerce/node-sass');
+const dartSass = require('sass');
+const Fiber = require('fibers');
+const { promisify } = require('util');
+
+class ScssCompiler {
+    /**
+     * WARNING! Make sure to create a new instance per each compilation to avoid processes interfering each other
+     *
+     * @param {object} logger
+     * @param {object} engines
+     * @param {object} engines.primary
+     * @param {object} engines.fallback
+     * @param {function} logger.error
+     */
+    constructor(
+        logger,
+        engines = {
+            primary: dartSass,
+            fallback: nodeSassFork,
+        },
+    ) {
+        this.logger = logger;
+        this.engines = engines;
+        this.files = {};
+        this.fullUrls = {};
+
+        this.activatePrimaryEngine();
```

Comment:
We activate primary here and also in the `compile` function

Reply:
Yeah, the reason was that other methods also rely on `this.engines.active` and when we run the tests of those separate methods they require `this.engines.active` to be set already. But at the same time in the `compile` method we wanna make sure that firstly we set the primary engine and then the fallback engine.

Reply:
Though, since this class is intended to be used only once per each compilation, we can just activate it in the constructor